java-client/src/main/java/com/cumulocity/sdk/client/RestConnector.java

```diff
@@ -211,24 +211,24 @@ public void postStream(String path, InputStream inputStream, MediaType inputStre
     }
 
     @Override
-    public <T extends ResourceRepresentation> T postFile(String path, T representation, byte[] bytes,
+    public <T extends ResourceRepresentation> T postFile(String path, T representation, byte[] bytes, MediaType mediaType,
```

Comment:
suggestion: Instead of creating more and more methods with more and more params, I would suggest to create param object like BinaryUpload that will carry all required parameters and can also set proper default values